src/compositions/TopBar/TopBar.tsx

```diff
@@ -1,34 +1,60 @@
-import React, { FC } from 'react';
+import React, { FC, useEffect } from 'react';
 
 import classNames from 'classnames';
 
 import Button from '../../components/Button/Button';
+import { truncateAddress } from '../../helpers/stringUtils';
+import useToggle from '../../hooks/useToggle';
 import { AppRoutes } from '../../routes';
 import IconButton from '../IconButton/IconButton';
 import IconNavLink from '../IconNavLink/IconNavLink';
+import UserPopup from '../UserPopup/UserPopup';
 import DesktopNav from './subcomponents/DesktopNav/DesktopNav';
 
 import './TopBar.scss';
 
 interface TopBarProps {
   mobileMenuIsVisible: boolean;
   showDesktopConnectButton: boolean;
+  showDesktopUserButton: boolean;
+  account: string | null | undefined;
+  ensAddress: string | undefined;
   onConnectButtonClick: () => void;
+  onDisconnectButtonClick: () => void;
   onMobileMenuButtonClick: () => void;
   className?: string;
 }
 
 const TopBar: FC<TopBarProps> = ({
   mobileMenuIsVisible,
   showDesktopConnectButton,
+  showDesktopUserButton,
+  account,
+  ensAddress,
   onConnectButtonClick,
+  onDisconnectButtonClick,
   onMobileMenuButtonClick,
   className = '',
 }) => {
+  const [isPopupOpen, toggleIsPopupOpen] = useToggle(false);
+
   const containerClassName = classNames('top-bar', {
     'top-bar--mobile-menu-is-visible': mobileMenuIsVisible,
   }, className);
 
+  const handleDisconnectClick = () => {
+    toggleIsPopupOpen();
+    onDisconnectButtonClick();
+  };
+
+  useEffect(() => {
+    if (isPopupOpen) {
```

Comment:
This will also trigger when you click inside of it. Only clicking outside should close it.